syntax = "proto3";

package db_serialization;

import "graph.proto";

message TransportRouterParameters {
    int32 bus_wait_time = 1;
    double bus_velocity = 2;
}

message PrevEdge {
    uint64 prev_edge = 1;
}

message RouteInternalData {
    double weight = 1;
    PrevEdge prev_edge_wrap = 2;
}

message VectorOfRouteInternalData {
    repeated RouteInternalData rid_vector = 1;
}

message RoutesInternalData {
    repeated VectorOfRouteInternalData all_router_data = 1;
}

message EdgeIdToBusName {
    uint64 edge_id = 1;
    string bus_name = 2;
}

message EdgeIdToSpanCount {
    uint64 edge_id = 1;
    int32 span_count = 2;
}

message TransportRouter {
    Graph graph = 1;
    RoutesInternalData internal_data = 2;
    repeated EdgeIdToBusName id_to_bus_name = 3;
    repeated EdgeIdToSpanCount id_to_span_count = 4;
}

